### PR TITLE
Add quick detail on meta labels.

### PR DESCRIPTION
**Pull request by mattdurham**
Add a very small bit of doc on meta labels, I was going to add instructions on how to keep them but this doesn't feel like the right place.

### CODE REVIEW

File: docs/sources/flow/reference/components/prometheus.remote_write.md [hunk 1]
```diff
@@ -332,3 +332,7 @@ prometheus.scrape "demo" {
 ## Compression
 
 `prometheus.remote_write` uses [snappy](https://en.wikipedia.org/wiki/Snappy_(compression)) for compression.
+
+## Meta labels
+
+Any labels that start with `__` will be removed before sending to the endpoint.
```
Comment (rfratto) [inline docs/sources/flow/reference/components/prometheus.remote_write.md:338]: Does this belong as a dedicated H2, or should it go closer to the documentation for where data is sent (the `endpoint` block)?

I'm concerned that adding too many h2s breaks the consistency patterns of how we document reference pages for components. 

cc @clayton-cornell

Comment (mattdurham) [reply docs/sources/flow/reference/components/prometheus.remote_write.md:338]: I am fine with that too, I think there is an unknown in where to put these little implementation details.

Comment (clayton-cornell) [reply docs/sources/flow/reference/components/prometheus.remote_write.md:338]: Could this be formatted as a note?

Comment (mattdurham) [reply docs/sources/flow/reference/components/prometheus.remote_write.md:338]: Is there a place for something like rolling up the compression/meta into this? Though it feels very grab bag. 

```
## Implementation details

* `prometheus.remote_write` uses [snappy](https://en.wikipedia.org/wiki/Snappy_(compression)) for compression.
* Any labels that start with `__` will be removed before sending to the endpoint.
```

Its almost like a FAQ or troubleshooting guidelike. The entry point might be why are my labels not being written to prometheus or what compression does remote_write use

**Comment by mattdurham** [comment]
I added technical details has a catchall. I dont think the name is great and very open to ideas though its something we can workshop later if we dont find a concensus.

Also added special labels

File: docs/sources/flow/reference/components/prometheus.scrape.md [hunk 2]
```diff
@@ -256,6 +256,19 @@ http://blackbox-exporter:9115/probe?target=grafana.com&module=http_2xx
 http://blackbox-exporter:9116/probe?target=grafana.com&module=http_2xx
 ```
 
-## Compression
+### Technical details
 
 `prometheus.scrape` supports [gzip](https://en.wikipedia.org/wiki/Gzip) compression.
+
+Special Labels that can change the behavior of prometheus.scrape
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.scrape.md:264]:
```suggestion
The following special labels can change the behavior of prometheus.scrape:
```

File: docs/sources/flow/reference/components/prometheus.scrape.md [hunk 3]
```diff
@@ -256,6 +256,19 @@ http://blackbox-exporter:9115/probe?target=grafana.com&module=http_2xx
...
+* `__address__` is the name of the label that holds the address of a scrape target.
+* `__metrics_path__`   is the name of the label that holds the path on which to scrape a target.
+* `__scheme__` is the name of the label that holds the scheme (http,https) on which to  scrape a target.
+* `__scrape_interval__` is the name of the label that holds the scrape interval used to scrape a target.
+* `__scrape_timeout__` is the name of the label that holds the scrape timeout used to scrape a target.
+* `__param__` is a prefix for labels that provide URL parameters used to scrape a target.
+
+* Special labels added after a scrape
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.scrape.md]:
```suggestion
The following special labels are added after a scrape:
```

**Review by clayton-cornell — COMMENTED** [review]
Some suggestions for formatting/syntax

File: docs/sources/flow/reference/components/prometheus.scrape.md [hunk 4]
```diff
@@ -257,6 +257,19 @@ http://blackbox-exporter:9115/probe?target=grafana.com&module=http_2xx
 http://blackbox-exporter:9116/probe?target=grafana.com&module=http_2xx
 ```
 
-## Compression
+### Technical details
 
 `prometheus.scrape` supports [gzip](https://en.wikipedia.org/wiki/Gzip) compression.
+
+The following special labels can change the behavior of prometheus.scrape:
+* `__address__` is the name of the label that holds the address of a scrape target.
```
Comment (tpaschalis) [inline docs/sources/flow/reference/components/prometheus.scrape.md:265]:
```suggestion
* `__address__` is the name of the label that holds the `<host>:<port>`  address of a scrape target.
```

WDYT? It'd tripped me up in the past, may be helpful here.

**Review by tpaschalis — APPROVED** [review]

File: docs/sources/flow/reference/components/prometheus.scrape.md (hunk 4 above)
Comment (clayton-cornell) [reply docs/sources/flow/reference/components/prometheus.scrape.md:265]: Making it clear exactly what address... a good idea. (side thought: Are there other areas where being more explicit like thsi will help?)

There's an extra space between port and address.

File: docs/sources/flow/reference/components/prometheus.remote_write.md (hunk 1 above)
Comment (clayton-cornell) [reply docs/sources/flow/reference/components/prometheus.remote_write.md]: I've been thinking about this, and don't yet have a suggestion.

* What would a user be doing with Agent where they'd need to know this info? Initial config? Fine tuning? Trying to fix a problem with data flow/collection?
* Is this simply nice to know info? or more critical to a successful Agent setup?

Comment (mattdurham) [reply docs/sources/flow/reference/components/prometheus.remote_write.md]: Its a common gotcha that people remark on. Usually like where are my labels? The compression one is more from a customer asking details to make sure we arent writing 5x the data we should.

Comment (clayton-cornell) [reply docs/sources/flow/reference/components/prometheus.remote_write.md]: @mattdurham This PR is getting old. I suggest we merge it as-is even though it's not 100% what we'd like. Get the info at least published and we can revisit this as it evolves.

**Pull request merged**